Rest sources controller

VolkerHBZ opened the pull request:
Bugfixes & Source Controller

VolkerHBZ committed:
rearranged HQL in UserController to adjust roleId, curatoryGroupId and status to their new semantics

VolkerHBZ committed:
Trouble with put/patch 404 issue

VolkerHBZ committed:
Trouble with put/patch 404 issue - fix

VolkerHBZ committed:
organizing test data to achieve determinism in running tests standalone as well as in a test suite.

VolkerHBZ committed:
ignoring the test for /register

VolkerHBZ committed:
CuratoryGroup search in UsersController repaired
Logging adjusted
UserProfileService.update test repaired

VolkerHBZ committed:
putting JSON imput data in a "data:" field, as specified

VolkerHBZ committed:
SourcesController + Tests

hornmo committed:
fix merge conflicts

hornmo commented on server/gokbg3/grails-app/services/org/gokb/UserProfileService.groovy:
```diff
@@ -79,19 +80,18 @@ class UserProfileService {
     def error = [:]
-    log.debug("Updating user ${user.id} ..")
-    def immutables = ['id', 'username', 'passwordExpired', 'last_alert_check']
+    log.debug("Updating user ${user.id} ..".toString())
```
We generally should not need to convert GStrings manually for logging.

hornmo commented on server/gokbg3/grails-app/services/org/gokb/UserProfileService.groovy:
```diff
@@ -79,19 +80,18 @@ class UserProfileService {
 
-    def reqBody = data
     if (!adminUser.isAdmin() && user != adminUser) {
       error.message = "$adminUser.username is not allowed to change $user.username"
```
A better solution would be to have an array of errors and adding objects like `[message: "$field not found", badData: ["${field}": $value], code: 400]` for each one.

hornmo commented on server/gokbg3/src/integration-test/groovy/gokbg3/rest/UsersTestSpec.groovy:
```diff
@@ -98,35 +115,37 @@ class UsersTestSpec extends AbstractAuthSpec {
   void "test PUT /rest/users/{id}"() {
     def urlPath = getUrlPath()
     // use the bearerToken to write to /rest/user
+    String bodyText = "{data:{id: $altUser.id," +
```
Preferably just define a map here and convert it via `body(bodyText as JSON)` later (like records in IntegrationControllerSpec for example).

hornmo reviewed (commented):
We don't really want to make calls from a service to a controller, as in UserDetailsService.update() to UserController.collectUserProp(). Therefore, collectUserProp should probably be moved to the service class.

VolkerHBZ committed:
applying the suggested changes:
- putting errors in an array within the JSON response
- moving collectUserProps to UserProfileService
- generating test JSON from a Container rather than using text

VolkerHBZ committed:
removing reference to UsersController from UserProfileService

VolkerHBZ committed:
adding data container to Post/rest/users request and test

VolkerHBZ committed:
Merge branch 'fix_#142' into rest_sourcesController

# Conflicts:
#	server/gokbg3/grails-app/controllers/org/gokb/rest/UsersController.groovy
#	server/gokbg3/grails-app/services/org/gokb/UserProfileService.groovy
#	server/gokbg3/src/integration-test/groovy/gokbg3/rest/ProfileTestSpec.groovy
#	server/gokbg3/src/integration-test/groovy/gokbg3/rest/UsersTestSpec.groovy

VolkerHBZ committed:
merging with changes from fix #142 branch

hornmo committed:
use updateObject in save
